### PR TITLE
Created initial version of retrieval script for postgresql jdbc drivers

### PR DESCRIPTION
**Pull request by soenkeliebau**
Not sure about the way I go about checking the sha1 sums, maybe that could be done without changing the files ..

Also, instructions for key retrieval for the gpg part need to be updated.

example:
./upload_new_postgres_driver_version.sh 42.2.9.jre7 test

# Description

*Please add a description here.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [ ] Changes are OpenShift compatible
- [ ] All added packages (via microdnf or otherwise) have a comment on why they are added
- [ ] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [ ] All packages should have (if available) signatures/hashes verified
- [ ] Does your change affect an SBOM? Make sure to update all SBOMs
```

### CODE REVIEW

File: hive/upload_new_postgres_driver_version.sh [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  echo "quitting"
+}
+
+# register the cleanup function to be called on the EXIT signal
+trap cleanup EXIT
+
+cd "$WORK_DIR" || exit
+
+bin_file=postgresql-$VERSION.jar
+src_file=postgresql-$VERSION-sources.jar
```
Comment (lfrancke) [inline hive/upload_new_postgres_driver_version.sh:34]: Do we need the sources?

File: hive/upload_new_postgres_driver_version.sh [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+echo '--> NOTE: Make sure you have downloaded and added the KEYS file (https://archive.apache.org/dist/hadoop/common/KEYS) to GPG: https://www.apache.org/info/verification.html'
+
+ if ! (gpg --verify "$bin_file.asc" "$bin_file" 2> /dev/null && gpg --verify "$src_file.asc" "$src_file" 2> /dev/null); then
+  echo "ERROR: One of the signatures could not be verified"
+  exit 1
+ fi
+
+echo "Uploading everything to Nexus"
+EXIT_STATUS=0
+curl --fail -u "$NEXUS_USER:$NEXUS_PASSWORD" --upload-file "$bin_file" 'https://repo.stackable.tech/repository/packages/hive/' || EXIT_STATUS=$?
```
Comment (lfrancke) [inline hive/upload_new_postgres_driver_version.sh:71]: Should this go to hive? I'd think somewhere else

File: hive/upload_new_postgres_driver_version.sh (hunk 1 above)
Comment (maltesander) [reply hive/upload_new_postgres_driver_version.sh]: If i my cite @soenkeliebau: "Better to have them than not to have them!"

File: hive/upload_new_postgres_driver_version.sh (hunk 2 above)
Comment (maltesander) [reply hive/upload_new_postgres_driver_version.sh]: i edited the script a little and opted for pgjdbc folder...we may use it somewhere else as well.

File: hive/upload_new_postgres_driver_version.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+# if ! (sha512sum --tag "$bin_file" | diff - "$bin_file.sha512" && sha512sum --tag "$src_file" | diff - "$src_file.sha512"); then
+if ! (sha1sum -c $bin_file.sha1 && sha1sum -c $src_file.sha1); then
+  echo "ERROR: One of the SHA1 sums does not match"
+  exit 1
+fi
+
+echo "Validating signatures"
+
+# TODO: check where to obtain keys used for signing these and fix the comment below
+echo '--> NOTE: Make sure you have downloaded and added the KEYS file (https://archive.apache.org/dist/hadoop/common/KEYS) to GPG: https://www.apache.org/info/verification.html'
```
[the remark this answers is not in the document]
Comment (maltesander) [reply hive/upload_new_postgres_driver_version.sh]: Correct keys file now referenced https://github.com/stackabletech/docker-images/pull/454/commits/0f14043e57ccd204a80ac2e467b90b467a8ebcf9

**Review by lfrancke — APPROVED** [review]

**Pull request merged**